### ISSUE TITLE
Using ArrayInput instead of StringInput breaks sub-command arguments

### ISSUE THREAD

**Issue by mcaskill** [mnapoli/silly#69]
Tested in mnapoli/silly v1.8.2, with [symfony/console](https://github.com/symfony/console) v6.2.5, via [laravel/valet](https://github.com/laravel/valet) v4.1.2.

While working on a fix for Valet, I've discovered that sub-command arguments no longer work as a result of #66. A [comment on #17](https://github.com/mnapoli/silly/issues/17#issuecomment-1597842322) appears to be the same issue I've discovered.

For context, Valet has among others the following commands [`link`](https://github.com/laravel/valet/blob/v4.1.2/cli/app.php#L193-L208) and [`isolate`](https://github.com/laravel/valet/blob/v4.1.2/cli/app.php#L632-L649). When executing the `link` command, if the `--isolate` argument is provided, Valet will run the `isolate` command via Silly's [`Application::runCommand` method](https://github.com/mnapoli/silly/blob/1.8.2/src/Application.php#L166C1-L173C6):

```php
$app->command('link [name] [--secure] [--isolate]', function ($name, $secure, $isolate) {
    # ...

    if ($isolate) {
        if (Site::phpRcVersion($name)) {
            $this->runCommand('isolate');
        } else {
```

If a custom name is provided for the link, it is not passed along to the `isolate` command which ends up using the current directory's name, which ends up causing an error because it won't find a matching link:

```php
$app->command('isolate [phpVersion] [--site=]', function (OutputInterface $output, $phpVersion, $site = null) {
    if (! $site) {
        $site = basename(getcwd());
    }
```

This is what I'm attempting to fix in Valet. The solution to this is fairly simple: provide the `--site=` option to the `isolate` sub-command with the given link name:

```diff
 if (Site::phpRcVersion($name)) {
-    $this->runCommand('isolate');
+    $this->runCommand('isolate --site="'.$name.'"');
 } else {
```

Unfortunately, this does not work. The `isolate` command _is_ executed but with _no_ arguments.

In Silly's `Application::runCommand()` method, the return value of `$input->getArguments()` is an empty array. The reason the arguments are empty is because the input is only parsed if it has a definition. The definition is assigned to the input by the command in Symfony's [`Command::run()` method](https://github.com/symfony/console/blob/v6.2.5/Command/Command.php#L270-L276).

The [Symfony documentation](https://symfony.com/doc/6.2/console/calling_commands.html), cited in #66, shows that `ArrayInput` is used for explicitly defined arguments. Here our arguments are defined via a string input (of tokens) which need a definition to be correctly parsed as arguments.

Reading again the quoted error message in #66, its messaging is suspiciously formulated:

> Too many arguments to "xxx" command, expected arguments "yyy".

Looking up the logic behind it in Symfony's [`ArgvInput::parseArgument()` method](https://github.com/symfony/console/blob/v6.2.5/Input/ArgvInput.php#L178-L184) gives me the impression their sub-command only accepts a single argument but they are passing multiple arguments (maybe an unquoted string). That's what triggers the above error message. By adding `new ArrayInput($input->getArguments()`, you are skipping all arguments which "fixes" that error message.

In conclusion, Silly should revert the changes introduced by #66 in v1.8.1:

```diff
- return $command->run(new ArrayInput($input->getArguments()), $output ?: new NullOutput());
+ return $command->run($input, $output ?: new NullOutput());
```

The sub-command will correctly parse the given arguments from the string input.
